src/lazy/text/buffer.rs

```diff
@@ -1991,7 +2047,12 @@ impl<'data> Stream for TextBuffer<'data> {
     }
 
     fn reset(&mut self, checkpoint: &Self::Checkpoint) {
+        let current_row = self.row;
+        let prev_column_value = self.prev_newline_offset;
+
         *self = *checkpoint;
+        self.row = current_row;
+        self.prev_newline_offset = prev_column_value;
```

Comment:
Can you explain what's going on here? If we're reseting to a specific checkpoint, why would we _not_ want to reset the row/col accordingly?

Reply:
Also, is it feasible to add/update a unit test for this function to include the row/col reset behavior?

Reply:
My understanding of why this is required: 
https://github.com/amazon-ion/ion-rust/pull/905#discussion_r1949866069